forms/HtmlEditorField.php

```diff
@@ -561,6 +564,48 @@ public function viewfile($request) {
 	}
 
 	/**
+	 * Find actions available on the given page. Attempt to exclude forms by getting lowercase actions only.
+	 *
+	 * @return array
+	 */
+	public function getactions() {
+		$id = (int)$this->getRequest()->getVar('PageID');
+		$actions = array();
+
+		if (($page = Page::get()->byID($id)) && !empty($page)) {
+			if (!$page->canView()) {
+				throw new SS_HTTPResponse_Exception(
+					_t(
+						'HtmlEditorField.ANCHORSCANNOTACCESSPAGE',
+						'You are not permitted to access the content of the target page.'
+					),
+					403
+				);
+			}
+
+			// Get lowercase actions
+			// ToDo: look up controller instead of assuming use of _Controller naming convention
```

Comment:
I could use some help with this part... I assume Director is able to work out what controller to use for a given class name but I couldn't find where the magic happens.